lib/migrate_assets_to_asset_manager.rb

```diff
@@ -26,6 +28,10 @@ def to_s
     "Migrating #{pluralize(@relative_file_paths.size, 'file')}"
   end
 
+  def self.directory_number_from_attachment_dir(attachment_dir)
+    attachment_dir.each_filename.entries.last.to_i
```

Comment:
Minor: Given we're confident that all directories are integers, I might use `Integer(attachment_dir.each_filename.entries.last)` rather than `String#to_i` so that this line will fail fast if the directory is not an integer. Note that `"abc".to_i # => 0`.

Reply:
I found this line `attachment_dir.each_filename.entries.last` hard to understand.

As I understand it, `attachment_dir` at this point would be something along the lines of `system/uploads/attachment_data/file/100/filename.pdf`. Is that right? I assume the `.each_filename.entries` bit is somehow obtaining `'100'` from this, but it's not at all obvious how this is working.

It's not obvious to me that this code is tested, so I think it would be useful to do one of the following: (a) add some tests; (b) extract explanatory local variables; or (c) explain how this works in the commit note.

Reply:
> As I understand it, attachment_dir at this point would be something along the lines of system/uploads/attachment_data/file/100/filename.pdf. Is that right?

No, i think it would be `system/uploads/attachment_data/file/100` due to the call to `glob('system/uploads/attachment_data/file/*')` in the block argument above. If the path was a string I'd do `.split('/').last` but `Pathname#split` doesn't work like `String#split` so I'd have to convert it to a String first, which is also confusing I think. I think the expectation test I added does test this method implicitly. I'll see if I can make it more obvious. 

The change to `Integer()` is a good idea - I'll do that.